editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -1955,7 +1955,7 @@ void CanvasItemEditor::_gui_input_viewport(const Ref<InputEvent> &p_event) {
 						real_t angle = node->get_rotation();
 						node->set_rotation(snap_angle(angle + (dfrom - center).angle_to(dto - center), angle));
```

Comment:
This line also needs to be fixed:

```
Vector2 pivot_offset = node->get_pivot_offset();
node->set_rotation(snap_angle(angle + (dfrom - (center + pivot_offset)).angle_to(dto - (center + pivot_offset)), angle));
```

This will fix the issue above.